iocore/hostdb/HostDB.cc

```diff
@@ -1607,7 +1608,8 @@ HostDBContinuation::remove_trigger_pending_dns()
     c = n;
   }
   while ((c = qq.dequeue())) {
-    c->handleEvent(EVENT_IMMEDIATE, nullptr);
+    // resume all queued HostDBCont in the thread associated with the netvc to avoid locking issues.
+    eventProcessor.schedule_imm(c);
```

Comment:
Presumably, most of the time, the old c->handleEvent() does get handled on the "right" thread right now? Otherwise, this would cause more problems than we see today? If so, is there not a way to check if we're currently on an "acceptable" thread, and conditionally do c->handleEvent()  if so ? I haven't looked at what we have available here, but it would be nice to avoid this rescheduling for the common case, and only take these rescheduling costs when absolutely necessary.

Also, is this related to the global session pools at all? It seems a lot of work is being done in the core, to address issues that only happens when the global session pool is enabled. My real concern here is changing all these things in the event system, when maybe we should revisit the global session pool (like, maybe understand why it's needed, or perhaps see if it can be improved / fixed?).

Reply:
@zwoop I'll rejigger the remove_trigger_pending_dns to only schedule if the affinity thread is different from the current thread.  That should calls the handleEvent to be called directly in the common case.  This is the only place in this PR that replaced a direct handleEvent call with a schedule.

The issue here has nothing to do with the global session pool.  It has to do with multiple hostdb requests queuing up for an outstanding DNS request.

Reply:
Pushed up a new commit with the rejigger